scripts/codemod/index.js

```diff
@@ -0,0 +1,199 @@
+/**
+ * Copyright (c) Meta Platforms, Inc. and affiliates.
+ *
+ * This source code is licensed under the MIT license found in the
+ * LICENSE file in the root directory of this source tree.
+ *
+ * @flow
+ */
+
+import type {ESNode, CallExpression} from 'hermes-estree';
+import type {TransformContext} from 'hermes-transform';
+
+const {transform, t} = require('hermes-transform');
+const {SimpleTraverser} = require('hermes-parser');
+const Glob = require('glob');
+const {readFileSync, writeFileSync} = require('fs');
+const Prettier = require('prettier');
+
+/* eslint-disable no-for-of-loops/no-for-of-loops */
+
+function containsReactDOMRenderCall(func: ESNode): boolean {
+  if (
+    func.type !== 'ArrowFunctionExpression' &&
+    func.type !== 'FunctionExpression'
+  ) {
+    throw new Error('expected a function');
+  }
+  let result = false;
+  SimpleTraverser.traverse(func.body, {
+    enter(node: ESNode) {
+      if (
+        node.type === 'CallExpression' &&
+        node.callee.type === 'MemberExpression' &&
+        node.callee.object.type === 'Identifier' &&
+        node.callee.object.name === 'ReactDOM' &&
+        node.callee.property.type === 'Identifier' &&
+        node.callee.property.name === 'render'
+      ) {
+        result = true;
+        throw SimpleTraverser.Break;
+      }
+    },
+    leave() {},
+  });
+  return result;
+}
+
+function updateItToAsync(context: TransformContext) {
+  return {
+    CallExpression(node: CallExpression) {
+      if (
+        node.callee.type === 'Identifier' &&
+        node.callee.name === 'it' &&
+        node.arguments.length === 2
+      ) {
+        const fn = node.arguments[1];
+        if (
+          fn.type !== 'ArrowFunctionExpression' &&
+          fn.type !== 'FunctionExpression'
+        ) {
+          throw new Error('expected a function as argument to it()');
+        }
+        if (containsReactDOMRenderCall(fn)) {
+          context.replaceNode(
+            fn,
+            t.ArrowFunctionExpression({
+              params: [],
+              body: fn.body,
+              async: true,
+            }),
+          );
+        }
+      }
+    },
+  };
+}
+
+// function updateExpectToAsync(context: TransformContext) {
```

Comment:
Are these commented out steps expected to be toggle on and used at all? Should we clean up for now and merge this so its easier to use it in the test migrations?